src/effects/native/flangereffect.cpp

```diff
@@ -84,22 +85,26 @@ void FlangerEffect::processChannel(const ChannelHandle& handle,
     Q_UNUSED(handle);
     Q_UNUSED(enableState);
     Q_UNUSED(groupFeatures);
-    Q_UNUSED(sampleRate);
-    CSAMPLE lfoPeriod = m_pPeriodParameter->value();
-    CSAMPLE lfoDepth = m_pDepthParameter->value();
-    // Unused in EngineFlanger
-    // CSAMPLE lfoDelay = m_pDelayParameter ?
-    //         m_pDelayParameter->value().toDouble() : 0.0f;
 
-    // TODO(rryan) check ranges
-    // period needs to be >=0
-    // delay needs to be >=0
-    // depth is ???
+    // TODO: remove assumption of stereo signal
+    const int kChannels = 2;
+
+    // The parameter minimum is zero so the exact center of the knob is 2 beats.
+    CSAMPLE periodTime = std::max(m_pPeriodParameter->value(), 0.05);
+    CSAMPLE lfoPeriod;
+    if (m_pSyncParameter->toBool() && groupFeatures.has_beat_length) {
+        // period is a number of beats
+        lfoPeriod = periodTime * groupFeatures.beat_length;
```

Comment:
Why do you need the file sample rate here? I think the track is already re-sampled here.